modules/_node-scripts/util/writeTimings.mjs

```diff
@@ -0,0 +1,39 @@
+import {constants} from 'fs';
+import * as fs from 'fs/promises';
+import * as path from 'path';
+
+import getRootDir from './getRootDir.mjs';
+
+export default async function writeTimings(start, endConfig) {
+	const {LIFERAY_NPM_SCRIPTS_TIMING} = process.env;
+
+	if (LIFERAY_NPM_SCRIPTS_TIMING) {
+		const end = Date.now();
+
+		const projectDir = path.relative(await getRootDir(), process.cwd());
+
+		const csvFilePath = path.join(LIFERAY_NPM_SCRIPTS_TIMING, `${projectDir}.csv`);
+
+		console.log(`Appending timing information to: ${csvFilePath}`);
+
+		try {
+			await fs.access(csvFilePath, constants.F_OK);
+		}
+		catch(error) {
+			if (error.code !== 'ENOENT') {
+				throw error;
+			}
+
+			await fs.mkdir(path.dirname(csvFilePath), {recursive: true});
+			await fs.writeFile(csvFilePath, `label,start,end,delta\n`, 'utf-8');
+		}
+
+		await fs.appendFile(
```

Comment:
One other random thought, can we write this to a .gitignore'd place? That way we could locally always have this enabled and check when needed

Reply:
This is limited by what npmscripts is doing now. This piggybacks on it, adding some more info to the file npmscripts creates, so we are not free to do whatever we want here (unless we change npmscripts, of course).

In any case you can write the file wherever you want. You just need to export an env var named `LIFERAY_NPM_SCRIPTS_TIMING` poiniting to any folder you choose and the files will be written there. 

You can export that variable in your `.bashrc` file, for example, if you want to have timings always on. However it will slow down your build a bit, most probably.